index.js

```diff
@@ -28,29 +28,28 @@ const register = function(server, options) {
       memory: process.memoryUsage(),
       version: package.version
     };
-
     if (settings.envs.length) {
       output.envs = settings.envs.reduce((memo, varName) => {
         memo[varName] = process.env[varName];
         return memo;
       }, {});
     }
-
-    await async.each(settings.checks, async check => {
-      if (!check.name || !check.method) {
-        throw new Boom('Invalid check');
-      }
-
-      output[check.name] = await str2fn.execute(
-        check.method,
-        server.methods,
-        {
-          request,
-          options: check.options
+    await Promise.all(settings.checks.map(check =>
+      new Promise(async (resolve, reject) => {
```

Comment:
Do you need `new Promise` here? doesn't `async () => ` default return a new promise?